get.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/s3"
+)
+
+// GetFromS3URL returns an io.ReadCloser instance and the content length (size in bytes) for the given S3 URL,
+// in the format specified by URLStyle.
+// More information about s3 URL styles: https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
+// If the URL defines a region (if provided) or bucket different from the one configured in this client, an error will be returned.
+func (cli *Client) GetFromS3URL(rawURL string, style URLStyle) (io.ReadCloser, *int64, error) {
+	return cli.doGetFromS3URL(rawURL, style, nil)
+
+}
+
+// GetFromS3URLWithPSK returns an io.ReadCloser instance and the content length (size in bytes) for the given S3 URL,
+// in the format specified by URLStyle, using the provided PSK for encryption.
+// More information about s3 URL styles: https://docs.aws.amazon.com/AmazonS3/latest/dev/VirtualHosting.html
+// If the URL defines a region (if provided) or bucket different from the one configured in this client, an error will be returned.
+func (cli *Client) GetFromS3URLWithPSK(rawURL string, style URLStyle, psk []byte) (io.ReadCloser, *int64, error) {
+	return cli.doGetFromS3URL(rawURL, style, psk)
+}
+
+func (cli *Client) doGetFromS3URL(rawURL string, style URLStyle, psk []byte) (io.ReadCloser, *int64, error) {
+	logData := log.Data{
+		"raw_url":   rawURL,
+		"url_style": style.String(),
+	}
+
+	// Parse URL with the provided format style
+	s3Url, err := ParseURL(rawURL, style)
+	if err != nil {
+		return nil, nil, NewError(fmt.Errorf("error parsing url: %w", err), logData)
+	}
+
+	// Validate that URL and client bucket names match
+	if s3Url.BucketName != cli.bucketName {
+		logData["bucket_name"] = cli.bucketName
+		return nil, nil, NewError(errors.New("unexpected bucket name in url"), logData)
+	}
+
+	// Validate that URL and client regions match, if URL provides one
+	if len(s3Url.Region) > 0 && s3Url.Region != cli.region {
+		logData["region"] = cli.region
+		return nil, nil, NewError(errors.New("unexpected aws region in url"), logData)
+	}
+
+	if psk == nil {
+		return cli.Get(s3Url.Key)
+	}
+	return cli.GetWithPSK(s3Url.Key, psk)
+}
+
+// Get returns an io.ReadCloser instance for the given path (inside the bucket configured for this client)
+// and the content length (size in bytes).
+// They 'key' parameter refers to the path for the file under the bucket.
+func (cli *Client) Get(key string) (io.ReadCloser, *int64, error) {
+
+	input := &s3.GetObjectInput{
+		Bucket: aws.String(cli.bucketName),
+		Key:    aws.String(key),
+	}
+
+	result, err := cli.sdkClient.GetObject(input)
+	if err != nil {
+		return nil, nil, NewError(fmt.Errorf("error getting object from s3: %w", err), log.Data{
+			"bucket_name": cli.bucketName,
+			"s3_key":      key, // key is the s3 filename with path (it's not a cryptographic key)
+			"user_psk":    false,
+		})
+	}
+
+	return result.Body, result.ContentLength, nil
```

Comment:
I think it might be good to state in the Godoc that the caller is responsible for closing the returned `ReadCloser`.

I know that should be pretty obvious but it removes any doubt and sets a clear expectation on the caller.